### PR TITLE
#270 wasm runtime context

### PR DESCRIPTION
**Pull request by saitofun**
1. wasm contexts:
> database executor with its own schema;
> cache, impls KVStore interface, use memory or redis
> logger with namespace and applet
> chain client
> project env vars
> project, applet and resource
2. redefine `wasm.ABI` interface and `wasmtime.ExportFuns` impls it
3. wasmtime.Runtime

### CODE REVIEW

**Review by hunshenshi — APPROVED** [review]

File: pkg/modules/vm/wasmtime/instance_exports.go [hunk 1]
```diff
@@ -34,362 +21,176 @@ const (
...
-		logger conflog.Logger
-		cl     *ChainClient
-	}
+func NewExportFuncs(ctx context.Context, code []byte) (*ExportFuncs, error) {
+	ef := &ExportFuncs{}
 
-	ChainClient struct {
-		pvk   *ecdsa.PrivateKey
-		chain *ethclient.Client
+	rt, err := NewRuntime(code, ef)
```
Comment (Liuhaai) [inline pkg/modules/vm/wasmtime/instance_exports.go:27]: Why is `Runtime` created in `ExportFuncs`?

File: pkg/modules/vm/wasmtime/instance_exports.go [hunk 2]
```diff
@@ -34,362 +21,176 @@ const (
...
+	ef.cl, _ = wasm.ChainClientFromContext(ctx)
+
+	return ef, nil
+}
 
-func (ef *ExportFuncs) Log(c *wasmtime.Caller, logLevel, ptr, size int32) int32 {
-	membuf := c.GetExport("memory").Memory().UnsafeData(ef.store)
-	buf, err := read(membuf, ptr, size)
+type ExportFuncs struct {
+	*Runtime
```
Comment (Liuhaai) [inline pkg/modules/vm/wasmtime/instance_exports.go:43]: Why is this added?
